cosmic-applet-power/src/lib.rs

```diff
@@ -151,17 +164,22 @@ impl cosmic::Application for Power {
                 Command::none()
             }
             Message::Action(action) => {
-                let id = window::Id::unique();
-                self.action_to_confirm = Some((id, action, COUNTDOWN_LENGTH));
-                get_layer_surface(SctkLayerSurfaceSettings {
-                    id,
-                    keyboard_interactivity: KeyboardInteractivity::None,
-                    anchor: Anchor::all(),
-                    namespace: "dialog".into(),
-                    size: Some((None, None)),
-                    size_limits: Limits::NONE.min_width(1.0).min_height(1.0),
-                    ..Default::default()
-                })
+                // Ask for user confirmation of non-destructive actions only
+                if [PowerAction::Lock, PowerAction::Suspend].contains(&action) {
```

Comment:
slight refactor: you should be able to use matches!() instead here

Reply:
Good point :+1:. That is better https://github.com/pop-os/cosmic-applets/compare/51cc478073fb9cb889a1359f361c5d6af7f4f2fa..5113042a1c5e1a350307b94984ed1828c4875c93